### ISSUE TITLE
Build status checks hang when using using windows-2022 pool

### ISSUE THREAD

**Issue by BretJohnson** [actions/virtual-environments#4079]
### Description

I updated our AzDO pipleline YAML to use windows-2022 instead of windows-2019, as we need to use VS 2022 tooling now.

With that change, the pipeline hangs very early, just reporting that it can't get status info to report:
![image](https://user-images.githubusercontent.com/245892/133147047-db3fc6eb-c49f-49c0-8c6b-0de7308c6f67.png)

I tried multiple times today and it's consistent, hanging the pipeline when and only when configuring one of the stages (not all of them, just one) to use the windows-2022 pool. I left it for a couple of hours and it's still hung, so it seems to be permanently hung.

Please let me know if there are additional troubleshooting steps that can help better diagnose.
Here is our build, though I'm not sure if you have access: https://github.com/xamarin/UITools/pull/425

### Virtual environments affected

- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Current windows-2022 image

### Is it regression?

Not sure - maybe

### Expected behavior

Build should proceed, not hang.

### Actual behavior

Build pipeline hangs. See description.

### Repro steps

See description.

**Comment by Darleev** [comment]
Hello @BretJohnson,
Could you please provide a sample repository where it is possible to reproduce the issue?
We're looking forward to your reply.

**Comment by Darleev** [comment]
@BretJohnson We tried to reproduce the issue on our side, but with no luck. Could you please assist us with this?
We're looking forward to your reply.

**Comment by BretJohnson** [comment]
I figured out what was going on here, late last week. The pipeline needs to be authorized to access the pool. If it's not authorized, unfortunately the github status checks don't show a good error, just seeming to hang like I described, but when you go to the AzDO pipeline to view the build results it has a better error and has a button to authorize access.

As the windows-2022 is new, and the pipeline wasn't previously authorized (apparently), it resulted in this behavior, until authorized.

Closing, given the above.
